web/react/stores/channel_store.jsx

```diff
@@ -10,212 +10,264 @@ var ActionTypes = Constants.ActionTypes;
+    },
+    resetCounts: function(id) {
+        var cm = this._getChannelMembers();
+        for (var cmid in cm) {
+            if (cm[cmid].channel_id === id) {
+                var c = this.get(id);
+                if (c) {
+                    cm[cmid].msg_count = this.get(id).total_msg_count;
+                    cm[cmid].mention_count = 0;
+                }
+                break;
+            }
+        }
+        this._storeChannelMembers(cm);
+    },
+    getCurrentId: function() {
+        return this._currentId;
+    },
+    getCurrent: function() {
+        var currentId = this.getCurrentId();
+
+        if (currentId) {
+            return this.get(currentId);
+        } else {
+            return null;
+        }
+    },
+    getCurrentMember: function() {
+        var currentId = ChannelStore.getCurrentId();
+
+        if (currentId) {
+            return this.getAllMembers()[currentId];
+        } else {
+            return null;
+        }
+    },
+    setChannelMember: function(member) {
+        var members = this._getChannelMembers();
+        members[member.channel_id] = member;
+        this._storeChannelMembers(members);
+        this.emitChange();
+    },
+    getCurrentExtraInfo: function() {
+        var currentId = ChannelStore.getCurrentId();
+        var extra = null;
+
+        if (currentId) {
+            extra = this._getExtraInfos()[currentId];
+        }
+
+        if (extra == null) {
+            extra = {members: []};
+        }
+
+        return extra;
+    },
+    getExtraInfo: function(channelId) {
+        var extra = null;
+
+        if (channelId) {
+            extra = this._getExtraInfos()[channelId];
+        }
+
+        if (extra == null) {
+            extra = {members: []};
+        }
+
+        return extra;
+    },
+    _storeChannel: function(channel) {
+        var channels = this._getChannels();
+        var found;
+
+        for (var i = 0; i < channels.length; i++) {
+            if (channels[i].id === channel.id) {
+                channels[i] = channel;
+                found = true;
+                break;
+            }
+        }
+
+        if (!found) {
+            channels.push(channel);
+        }
 
-    return channels;
-  },
-  _storeExtraInfos: function(extraInfos) {
-    BrowserStore.setItem("extra_infos", extraInfos);
-  },
-  _getExtraInfos: function() {
-    return BrowserStore.getItem("extra_infos", {});
-  },
-  isDefault: function(channel) {
-    return channel.name == Constants.DEFAULT_CHANNEL;
-  }  
+        channels.sort(function chanSort(a, b) {
+            if (a.display_name.toLowerCase() < b.display_name.toLowerCase()) {
+                return -1;
+            }
+            if (a.display_name.toLowerCase() > b.display_name.toLowerCase()) {
+                return 1;
+            }
+            return 0;
+        });
+
+        this._storeChannels(channels);
+    },
+    _storeChannels: function(channels) {
```

Comment:
As part of the style guide, we can not have any functions that start with an underscore